lightning_covid19/model/base_densenet.py

```diff
@@ -0,0 +1,194 @@
+                acc = torch.mean(acc)
+
+            acc_mean += acc
+
+        loss_mean /= len(outputs)
+        acc_mean /= len(outputs)
+        tqdm_dict = {f'{prefix}_loss': loss_mean, f'{prefix}_acc': acc_mean}
+        result = {'progress_bar': tqdm_dict, 'log': tqdm_dict, f'{prefix}_loss': loss_mean}
+        return result
+
+    def configure_optimizers(self):
+        optimizer = optim.Adam(self.parameters(), lr=self.hparams.learning_rate)
+        scheduler = optim.lr_scheduler.CosineAnnealingLR(optimizer, T_max=10)
+        return [optimizer], [scheduler]
+
+    # ------------- Model hyperparameters with default values ------------- #
+
+    @staticmethod
+    def add_model_specific_args(parent_parser, root_dir):  # pragma: no cover
+        """
+        Parameters you define here are available to your model through self.hparams
+        :param parent_parser: parent_parser is set up typically in the main training script
+        :param root_dir:
+        :return:
+        """
+        parser = ArgumentParser(parents=[parent_parser])
+
+        # data
+        git_root = run('git rev-parse --show-toplevel').strip()
+        parser.add_argument('--data_root', default=f'{git_root}/data/covid-chestxray-dataset', type=str)
+
+        # nn & training params
+        parser.add_argument('--max_epochs', default=20, type=int)
+        parser.add_argument('--batch_size', default=4, type=int)
+        parser.add_argument('--learning_rate', default=0.001, type=float)
+        parser.add_argument('--xraysize', default=224, type=int, help='Do no change unless you change covid dataset')
+        return parser
+
+    # --------------------- Data preparation --------------------- #
+
+    def prepare_data(self):
+        chestxray_root = self.hparams.data_root
+        if os.path.exists(chestxray_root):
+            assert os.path.isdir(chestxray_root), f'{chestxray_root} should be cloned from {clone_uri}'
+        else:
+            print('Cloning the covid chestxray dataset. It may take a while\n...\n', flush=True)
+            run(f'git clone https://github.com/ieee8023/covid-chestxray-dataset.git {chestxray_root}')
+        transform = transforms.Compose([xrv.datasets.XRayCenterCrop(),
+                                        xrv.datasets.XRayResizer(self.hparams.xraysize)])
+        covid19 = xrv.datasets.COVID19_Dataset(
+            imgpath=f'{chestxray_root}/images',
+            csvpath=f'{chestxray_root}/metadata.csv',
+            transform=transform)
+        print(f'Covid Chest x-ray stats dataset stats:\n{covid19}\n\n', flush=True)
+
+        # count split sizes
+        n_train= int(0.8 * len(covid19))
+        n_valid_test = len(covid19) - n_train
+        n_valid = int(0.5 * n_valid_test)
+        n_test = n_valid_test - n_valid
+
+        # split the dataset
+        self.train_set, self.val_set, self.test_set = \
+            torch.utils.data.random_split(covid19, [n_train, n_valid, n_test])
```

Comment:
Possible data leakage?
On the original dataset, there are several images from the same patient [see for example patient number 2](https://github.com/ieee8023/covid-chestxray-dataset/blob/master/metadata.csv#L2-L5) 
Should we take this into account when splitting the data?

Reply:
Great catch! We definitely should..... 

The scenario with covid19 is that a completely new person is diagnosed.
It is hardly acceptable to assume that our model has seen the xray image of that new person before.

Let me try to find a solution quickly - otherwise, I need to hand it over to somebody I am bus today.

Reply:
Unfortunately not a simple fix without changing torchxrayvision https://github.com/mlmed/torchxrayvision/issues/2

I opened https://github.com/PyTorchLightning/lightning-Covid19/issues/11 so we won't forget on that - I suggest to keep it as a bug and accept this MR.

@Borda thoughts?